### PR TITLE
Implemented tests and validation for P2MS

### PR DESCRIPTION
**Pull request by supreme2580**
- [x] issue #
- [x] follows contribution [guide](https://github.com/keep-starknet-strange/shinigami/blob/main/CONTRIBUTING.md)
- [x] code change includes tests

Closes: https://github.com/keep-starknet-strange/shinigami/issues/248

### CODE REVIEW

File: packages/engine/src/tests/test_p2ms.cairo [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    let utxo_hints = array![prev_out];
+
+    let res = validate::validate_p2ms(@transaction, 0, utxo_hints);
+    match res {
+        Result::Ok(()) => {},
+        Result::Err(err) => panic!("P2MS 1-of-2 transaction validation failed: {}", err),
+    }
+}
+
+// #[test]
```
Comment (b-j-roberts) [inline packages/engine/src/tests/test_p2ms.cairo:29]: This test is currently not running, I think

File: packages/engine/src/tests/test_p2ms.cairo [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    let raw_transaction = hex_to_bytecode(@raw_transaction_hex);
+    let transaction = TransactionTrait::deserialize(raw_transaction);
+    let utxo_hints = array![prev_out];
+
+    let res = validate::validate_p2ms(@transaction, 0, utxo_hints);
+    assert!(res.is_ok(), "P2MS 3-of-3 transaction validation failed");
+}
+
+#[test]
+fn test_p2ms_edge_cases() {
```
Comment (b-j-roberts) [inline packages/engine/src/tests/test_p2ms.cairo:70]: Which transaction is this testing?

**Review by b-j-roberts — CHANGES_REQUESTED** [review]
Nice work so far!

File: packages/engine/src/tests/test_p2ms.cairo (hunk 1 above)
Comment (supreme2580) [reply packages/engine/src/tests/test_p2ms.cairo:29]: Will confirm this and get back to you

File: packages/engine/src/tests/test_p2ms.cairo (hunk 2 above)
Comment (supreme2580) [reply packages/engine/src/tests/test_p2ms.cairo:70]: @b-j-roberts This is to test cases of 1 of 1 multisig, 1 of 2 multisig with one of the keys being invalid and 2-of-3 multisig with maximum allowed public keys (20)

Comment (b-j-roberts) [reply packages/engine/src/tests/test_p2ms.cairo:70]: Can you add the txids to the script? If they are 3 separate transactions, they should all be different tests too and run validate 3 times ( one for each ).

Comment (supreme2580) [reply packages/engine/src/tests/test_p2ms.cairo:70]: Will split asap

Comment (supreme2580) [reply packages/engine/src/tests/test_p2ms.cairo]: @b-j-roberts I have split the edge cases into individual tests

File: packages/tests/src/tests/test_p2ms.cairo [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    let transaction = EngineInternalTransactionTrait::deserialize(raw_transaction);
+    let utxo_hints = array![prev_out];
+
+    let res = validate::validate_p2ms(@transaction, 0, utxo_hints);
+    assert!(res.is_ok(), "P2MS 3-of-3 transaction validation failed");
+}
+
+#[test]
+fn test_p2ms_1of1() {
+    // Test case for 1-of-1 multisig
```
Comment (b-j-roberts) [inline packages/tests/src/tests/test_p2ms.cairo:71]: Can you add the txid to this like the other tests above? Trying to verify this test

File: packages/tests/src/tests/test_p2ms.cairo [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    let transaction = EngineInternalTransactionTrait::deserialize(raw_transaction);
+    let utxo_hints = array![prev_out_1of1];
+
+    let res = validate::validate_p2ms(@transaction, 0, utxo_hints);
+    assert!(res.is_ok(), "P2MS 1-of-1 multisig validation failed");
+}
+
+#[test]
+fn test_p2ms_1of2_invalid_pubkey() {
+    // Test case for 1-of-2 multisig with an invalid public key
```
Comment (b-j-roberts) [inline packages/tests/src/tests/test_p2ms.cairo:92]: Can you add the txid to this as well?

File: packages/tests/src/tests/test_p2ms.cairo [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+    let raw_transaction_hex =
+        "0x0100000001b14bdcbc3e01bdaad36cc08e81e69c82e1060bc14e518db2b49aa43ad90ba26000000000490047304402203f16c6f40162ab686621ef3000b04e75418a0c0cb2d8aebeac894ae360ac1e780220ddc15ecdfc3507ac48e1681a33eb60996631bf6bf5bc0a0682c4db743ce7ca2b01ffffffff0140420f00000000001976a914660d4ef3a743e3e696ad990364e555c271ad504b88ac00000000";
+    let raw_transaction = hex_to_bytecode(@raw_transaction_hex);
+    let transaction = EngineInternalTransactionTrait::deserialize(raw_transaction);
+    let utxo_hints = array![prev_out_1of2_invalid];
+
+    let res = validate::validate_p2ms(@transaction, 0, utxo_hints);
+    assert!(res.is_err(), "P2MS 1-of-2 with invalid public key should fail validation");
+}
```
Comment (b-j-roberts) [inline packages/tests/src/tests/test_p2ms.cairo:109]: There is also a 2-of-3 ms example and a 20-of-20 example that would be good to have tests for if you can.

**Review by b-j-roberts — CHANGES_REQUESTED** [review]

File: packages/tests/src/tests/test_p2ms.cairo (hunk 5 above)
Comment (supreme2580) [reply packages/tests/src/tests/test_p2ms.cairo:109]: Okay boss will add this

File: packages/tests/src/tests/test_p2ms.cairo (hunk 3 above)
Comment (supreme2580) [reply packages/tests/src/tests/test_p2ms.cairo:71]: Okay boss will add the txids

**Review by b-j-roberts — APPROVED** [review]
